File: fluentcore/src/main/java/com/mcreater/fxfluent/util/DefaultFontPatcher.java
package com.mcreater.fxfluent.util;

import javafx.scene.text.Font;

import java.lang.reflect.Field;

public class DefaultFontPatcher {
    public static void patch(Font font) throws NoSuchFieldException, IllegalAccessException {
        System.setProperty("prism.lcdtext", "false");
        Field field = Font.class.getDeclaredField("DEFAULT");
        field.setAccessible(true);
        field.set(null, font);
    }
}

File: fluentcore/src/main/java/com/mcreater/fxfluent/util/DefaultFontPatcher.kt
package com.mcreater.fxfluent.util

import javafx.scene.text.Font

class DefaultFontPatcher {
    companion object {
        @JvmStatic
        @Throws(NoSuchFieldException::class, IllegalAccessException::class)
        fun patch(font: Font?) {
            val field = Font::class.java.getDeclaredField("DEFAULT")
            field.setAccessible(true)
            field[null] = font
        }
    }
}